meson.build

```diff
@@ -0,0 +1,834 @@
+
+###########
+# symbols #
+###########
+
+symbols_sources = [
+    'misc/symbols.c',
+    'utils/argspec.c',
+    'utils/auto-args.c',
+    'utils/debug.c',
+    'utils/demangle.c',
+    'utils/dwarf.c',
+    'utils/filter.c',
+    'utils/rbtree.c',
+    'utils/regs.c',
+    'utils/session.c',
+    'utils/symbol-libelf.c',
+    'utils/symbol-rawelf.c',
+    'utils/symbol.c',
+    'utils/utils.c',
+]
+
+symbols_deps = [
+    cxa_demangle_dep,
+    libdl_dep,
+    libdw_dep,
+    libelf_dep,
+    libunwind_dep,
+]
+
+symbols = executable(
+    'symbols', [symbols_sources],
+    dependencies: [symbols_deps],
+    c_args: [common_cflags, symbols_cflags],
+    link_args: [common_ldflags],
+    build_by_default: true,
+)
+
+#########
+# tests #
+#########
+
+unittests_includes = include_directories('tests', 'arch/' + arch)
+
+unittests_sources = [
+    'cmds/dump.c',
+    'cmds/graph.c',
+    'cmds/info.c',
+    'cmds/live.c',
+    'cmds/record.c',
+    'cmds/recv.c',
+    'cmds/replay.c',
+    'cmds/report.c',
+    'cmds/script.c',
+    'cmds/tui.c',
+    'tests/unittest.c',
+    'uftrace.c',
+    'utils/argspec.c',
+    'utils/auto-args.c',
+    'utils/data-file.c',
+    'utils/debug.c',
+    'utils/demangle.c',
+    'utils/dwarf.c',
+    'utils/extern.c',
+    'utils/field.c',
+    'utils/filter.c',
+    'utils/fstack.c',
+    'utils/graph.c',
+    'utils/hashmap.c',
+    'utils/kernel.c',
+    'utils/pager.c',
+    'utils/perf.c',
+    'utils/rbtree.c',
+    'utils/regs.c',
+    'utils/report.c',
+    'utils/script-luajit.c',
+    'utils/script-python.c',
+    'utils/script.c',
+    'utils/session.c',
+    'utils/symbol-libelf.c',
+    'utils/symbol-rawelf.c',
+    'utils/symbol.c',
+    'utils/utils.c',
+    libmcount_arch_sources,
+    libmcount_sources,
+]
+
+unittests_deps = [
+    cxa_demangle_dep,
+    libcapstone_dep,
+    libdl_dep,
+    libdw_dep,
+    libelf_dep,
+    libluajit_dep,
+    libncursesw_dep,
+    libpthread_dep,
+    libpython_dep,
+    librt_dep,
+    libtraceevent_dep,
+    libunwind_dep,
+]
+
+tests_dir = meson.source_root() + '/tests'
+
+tests_unittest_h = tests_dir + '/unittest.h'
+
+unittests = executable(
+    'unittests', [unittests_sources],
+    include_directories: [unittests_includes, libmcount_arch_include],
+    dependencies: [unittests_deps],
+    c_args: [common_cflags, test_cflags,  '-DUNIT_TEST', '-include', tests_unittest_h],
+    link_args: [common_ldflags],
+    build_by_default: false,
+)
+
+run_target('unittest',
+    command: [unittests],
+    depends: [unittests],
+)
+
+run_target('runtest',
+    command: ['sh', '-c', 'cd @0@ && objdir=@1@ ./runtest.py'.format(tests_dir, meson.current_build_dir())],
+    depends: [
+        libmcount,
+        libmcount_fast,
+        libmcount_single,
+        libmcount_fast_single,
+        libmcount_fast_nop,
+        uftrace,
+    ],
+)
+
+run_target('tests',
+    command: ['sh', '-c', 'cd @0@ && ninja unittest; ninja runtest'.format(meson.current_build_dir())],
```

Comment:
My understanding of the `test()` feature of Meson is to setup all your test individually using this command. In Uftrace, all tests are part of a single binary. Hence, I don't see the point of using the `test()` feature. In other projects I did, tests were separate binaries and it made sense to use `test()`. Anyway, I think it would be a good idea in the future to eventually switch to this Meson feature.